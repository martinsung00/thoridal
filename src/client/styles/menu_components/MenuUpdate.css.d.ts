export const root: string;
export const description: string;
export const arrow: string;
export const arrowTop: string;
export const arrowBottom: string;
export const arrowTwo: string;
export const arrowThree: string;
export const mymodal: string;
export const myoverlay: string;
export const ReactModal__Overlay: string;
export const reactModalOverlay: string;
export const reactModalOverlayAfterOpen: string;
export const reactModalOverlayBeforeClose: string;
export const closeModal: string;
