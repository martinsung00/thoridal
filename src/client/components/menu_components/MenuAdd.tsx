import React, { useState } from "react";
import styles from "../../styles/menu_components/MenuAdd.css";
import Modal from "react-modal";
import ModalContent from "./modal_components/add/Add";

export const Add: React.FunctionComponent<{}> = () => {
  const [show, toggleShow] = useState(false);

  return (
    <div>
      <Modal
        isOpen={show}
        shouldCloseOnOverlayClick={true}
        onRequestClose={function (): void {
          /* istanbul ignore next */
          toggleShow(!show);
        }}
        contentLabel="My dialog"
        id="modal-component"
        className={show ? styles.mymodal : "display-none"}
        overlayClassName={styles.myoverlay}
        closeTimeoutMS={0}
      >
        <div
          onClick={function (): void {
            toggleShow(!show);
          }}
          className={styles.closeModal}
          id="close-modal"
        >
          {"\u00d7"}
        </div>
        <ModalContent toggleShow={toggleShow.bind(this)} show={show} />
      </Modal>
      <div
        className={styles.root}
        id="menu-add"
        onClick={function () {
          toggleShow(!show);
        }}
      >
        <div className={styles.description}>Write a Trade</div>
        <div className={styles.addLogo} />
        <div className={styles.decoration} />
        <div className={styles.decorationTop} />
      </div>
    </div>
  );
};
