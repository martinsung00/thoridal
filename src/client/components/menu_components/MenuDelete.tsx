import React, { useState } from "react";
import styles from "../../styles/menu_components/MenuDelete.css";
import Modal from "react-modal";
import ModalContent from "./modal_components/delete/Delete";

export const Delete: React.FunctionComponent<{}> = () => {
  const [show, toggleShow] = useState(false);

  return (
    <div>
      <Modal
        isOpen={show}
        shouldCloseOnOverlayClick={true}
        onRequestClose={function (): void {
          /* istanbul ignore next */
          toggleShow(!show);
        }}
        contentLabel="My dialog"
        id="modal-component"
        className={show ? styles.mymodal : "display-none"}
        overlayClassName={styles.myoverlay}
        closeTimeoutMS={0}
      >
        <div
          onClick={function (): void {
            toggleShow(!show);
          }}
          className={styles.closeModal}
          id="close-modal"
        >
          {"\u00d7"}
        </div>
        <ModalContent toggleShow={toggleShow.bind(this)} show={show}/>
      </Modal>
      <div
        className={styles.root}
        id="menu-delete"
        onClick={function () {
          toggleShow(!show);
        }}
      >
        <div className={styles.description}>Delete a Trade</div>
        <div className={styles.deleteLogo} />
        <div className={styles.decoration}>{"\u00d7"}</div>
      </div>
    </div>
  );
};
