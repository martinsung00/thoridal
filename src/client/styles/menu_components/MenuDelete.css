.root {
  height: 30.4em;
  width: 20%;
  position: relative;
  float: right;
  right: 6.5%;
  top: 0;
  margin-bottom: 5%;
  border: 5px solid rgb(133, 133, 133);
  border-radius: 3.5px;
  cursor: pointer;
  transition: 0.5s ease-out;
}

.root:hover {
  height: 30.4em;
  width: 20%;
  position: relative;
  float: right;
  right: 6.5%;
  top: 0;
  margin-bottom: 5%;
  border: 5px solid rgb(48, 48, 48);
  border-radius: 3.5px;
  cursor: pointer;
}

.description {
  position: relative;
  margin-top: 20%;
  font-family: Helvetica, Arial, sans-serif;
  font-size: 30px;
  text-align: center;
  color: rgb(133, 133, 133);
  transition: 0.5s ease-out;
}

.root:hover .description {
  position: relative;
  margin-top: 20%;
  font-family: Helvetica, Arial, sans-serif;
  font-size: 30px;
  text-align: center;
  color: rgb(48, 48, 48);
}

.deleteLogo {
  height: 11em;
  width: 11em;
  position: relative;
  margin-top: 42%;
  margin-left: 9%;
  border: 11px solid rgb(133, 133, 133);
  border-radius: 100%;
  display: block;
  transition: 0.5s ease-out;
}

.root:hover .deleteLogo {
  height: 11em;
  width: 11em;
  position: relative;
  margin-top: 42%;
  margin-left: 9%;
  border: 11px solid rgb(48, 48, 48);
  border-radius: 100%;
  display: block;
}

.decoration {
  height: 30%;
  width: 30%;
  position: relative;
  font-family: Helvetica, Arial, sans-serif;
  font-size: 210px;
  margin-top: -99.5%;
  margin-left: 24.65%;
  color: rgb(133, 133, 133);
  transition: 0.5s ease-out;
}

.root:hover .decoration {
  height: 30%;
  width: 30%;
  position: relative;
  font-family: Helvetica, Arial, sans-serif;
  font-size: 210x;
  margin-top: -99.5%;
  margin-left: 24.5%;
  color: rgb(255, 0, 0);
}

/* Modal Control */

.mymodal {
  position: fixed;
  width: 60%;
  height: 90%;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  border: 1px solid #ccc;
  background: #fff;
  overflow-y: scroll;
  overflow-x: hidden;
  border-radius: 4px;
  outline: none;
  padding: 20px;
  transition: opacity 1s ease-in;
}

.myoverlay {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: rgba(0, 0, 0, 0.75);
  outline: none;
  transition: opacity 1s ease-in;
}

.ReactModal__Overlay {
  opacity: 0;
  transition: opacity 500ms ease-in;
  outline: none;
}

.ReactModal__Overlay--after-open {
  opacity: 1;
  transition: opacity 1s ease-in;
  outline: none;
}

.ReactModal__Overlay--before-close {
  opacity: 0;
  transition: opacity 500ms ease-in;
  outline: none;
}

.closeModal {
  float: right;
  padding: none;
  padding-top: 2.2%;
  margin-right: 11px;
  margin-top: -5px;
  font-family: Helvetica, sans-serif;
  color: darkgray;
  cursor: pointer;
  outline: none;
  transition: opacity 500ms ease-in;
}

.closeModal:hover {
  float: right;
  padding: none;
  padding-top: 2.2%;
  margin-right: 11px;
  margin-top: -5px;
  font-family: Helvetica, sans-serif;
  color: black;
  cursor: pointer;
  outline: none;
  transition: opacity 500ms ease-in;
}
