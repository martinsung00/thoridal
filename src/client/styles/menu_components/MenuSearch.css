.root {
  height: 30.4em;
  width: 20%;
  position: relative;
  float: right;
  right: 3%;
  top: 0;
  margin-bottom: 5%;
  border: 5px solid rgb(133, 133, 133);
  border-radius: 3.5px;
  cursor: pointer;
  transition: 0.5s ease-out;
}

.root:hover {
  height: 30.4em;
  width: 20%;
  position: relative;
  float: right;
  right: 3%;
  top: 0;
  margin-bottom: 5%;
  border: 5px solid rgb(48, 48, 48);
  border-radius: 3.5px;
  cursor: pointer;
}

.description {
  position: relative;
  margin-top: 20%;
  font-family: Helvetica, Arial, sans-serif;
  font-size: 30px;
  text-align: center;
  color: rgb(133, 133, 133);
  transition: 0.5s ease-out;
}

.root:hover .description {
  position: relative;
  margin-top: 20%;
  font-family: Helvetica, Arial, sans-serif;
  font-size: 30px;
  text-align: center;
  color: rgb(48, 48, 48);
}

.searchLogo {
  height: 6.5em;
  width: 6.5em;
  position: relative;
  margin-top: 46%;
  margin-left: 17%;
  border: 14px solid rgb(133, 133, 133);
  border-radius: 100%;
  display: block;
  transition: 0.5s ease-out;
}

.root:hover .searchLogo {
  height: 6.5em;
  width: 6.5em;
  position: relative;
  margin-top: 46%;
  margin-left: 17%;
  border: 14px solid rgb(48, 48, 48);
  background-color: rgb(0, 193, 252);
  border-radius: 100%;
  display: block;
}

.decoration {
  height: 4em;
  width: 1em;
  position: relative;
  margin-top: -15%;
  margin-left: 68%;
  background-color: rgb(133, 133, 133);
  transform: rotate(-45deg);
  transition: 0.5s ease-out;
}

.root:hover .decoration {
  height: 4em;
  width: 1em;
  position: relative;
  margin-top: -15%;
  margin-left: 68%;
  background-color: rgb(48, 48, 48);
  transform: rotate(-45deg);
}

/* Modal Control */

.mymodal {
  position: fixed;
  width: 100%;
  height: 100%;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  border: 1px solid #ccc;
  background: #fff;
  overflow-y: scroll;
  overflow-x: hidden;
  border-radius: 4px;
  outline: none;
  padding: 20px;
  transition: opacity 1s ease-in;
}

.myoverlay {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: rgba(0, 0, 0, 0.75);
  outline: none;
  transition: opacity 1s ease-in;
}

.ReactModal__Overlay {
  opacity: 0;
  transition: opacity 500ms ease-in;
  outline: none;
}

.ReactModal__Overlay--after-open {
  opacity: 1;
  transition: opacity 1s ease-in;
  outline: none;
}

.ReactModal__Overlay--before-close {
  opacity: 0;
  transition: opacity 500ms ease-in;
  outline: none;
}

.closeModal {
  float: right;
  padding: none;
  padding-top: 2.2%;
  margin-right: 11px;
  margin-top: -5px;
  font-family: Helvetica, sans-serif;
  color: darkgray;
  cursor: pointer;
  outline: none;
  transition: opacity 500ms ease-in;
}

.closeModal:hover {
  float: right;
  padding: none;
  padding-top: 2.2%;
  margin-right: 11px;
  margin-top: -5px;
  font-family: Helvetica, sans-serif;
  color: black;
  cursor: pointer;
  outline: none;
  transition: opacity 500ms ease-in;
}
