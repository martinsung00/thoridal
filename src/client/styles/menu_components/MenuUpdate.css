.root {
  height: 30.4em;
  width: 20%;
  position: relative;
  float: right;
  right: 10.25%;
  top: 0;
  margin-bottom: 5%;
  border: 5px solid rgb(133, 133, 133);
  border-radius: 3.5px;
  cursor: pointer;
  transition: 0.5s ease-out;
}

.root:hover {
  height: 30.4em;
  width: 20%;
  position: relative;
  float: right;
  right: 10.25%;
  top: 0;
  margin-bottom: 5%;
  border: 5px solid rgb(48, 48, 48);
  border-radius: 3.5px;
  cursor: pointer;
}

.description {
  position: relative;
  margin-top: 20%;
  font-family: Helvetica, Arial, sans-serif;
  font-size: 30px;
  text-align: center;
  color: rgb(133, 133, 133);
  transition: 0.5s ease-out;
}

.root:hover .description {
  position: relative;
  margin-top: 20%;
  font-family: Helvetica, Arial, sans-serif;
  font-size: 30px;
  text-align: center;
  color: rgb(48, 48, 48);
}

.arrow {
  cursor: pointer;
  height: 120px;
  left: 33%;
  position: absolute;
  top: 38%;
  transform: rotate(-90deg);
  transition: transform 0.1s;
  width: 80px;
}

.arrowTop {
  background-color: rgb(133, 133, 133);
  height: 10px;
  left: -5px;
  position: absolute;
  top: 50%;
  width: 100%;
  transform: rotate(45deg);
  transform-origin: bottom right;
}

.arrowBottom {
  background-color: rgb(133, 133, 133);
  height: 10px;
  left: -5px;
  position: absolute;
  top: 47%;
  width: 100%;
  transform: rotate(-45deg);
  transform-origin: top right;
}

.arrowBottom:after {
  background-color: rgb(0, 185, 9);
  content: "";
  height: 100%;
  position: absolute;
  top: 0;
  transition: 0.15s;
  left: 0;
  right: 100%;
  transition-delay: 0.15s;
}

.arrowTop:after {
  background-color: rgb(0, 185, 9);
  content: "";
  height: 100%;
  position: absolute;
  top: 0;
  transition: 0.15s;
  left: 100%;
  right: 0;
  transition-delay: 0.15s;
}

.arrow:after {
  left: 0;
  right: 100%;
  transition-delay: 0.15s;
}

.root:hover .arrowTop:after {
  left: 0;
  transition-delay: 0.15s;
}

.root:hover .arrowBottom:after {
  right: 0;
  transition-delay: 0s;
}

.arrow:active {
  transform: rotate(-90deg) scale(0.9);
}

/* Second Arrow */

.arrowTwo {
  cursor: pointer;
  height: 120px;
  left: 33%;
  position: absolute;
  top: 47%;
  transform: rotate(-90deg);
  transition: transform 0.1s;
  width: 80px;
}

.arrowTwo:active {
  transform: rotate(-90deg) scale(0.9);
}

/* Third Arrow */

.arrowThree {
  cursor: pointer;
  height: 120px;
  left: 33%;
  position: absolute;
  top: 56%;
  transform: rotate(-90deg);
  transition: transform 0.1s;
  width: 80px;
}

.arrowThree:active {
  transform: rotate(-90deg) scale(0.9);
}

/* Modal Content */

.mymodal {
  position: fixed;
  width: 60%;
  height: 90%;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  border: 1px solid #ccc;
  background: #fff;
  overflow-y: scroll;
  overflow-x: hidden;
  border-radius: 4px;
  outline: none;
  padding: 20px;
  transition: opacity 1s ease-in;
}

.myoverlay {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: rgba(0, 0, 0, 0.75);
  outline: none;
  transition: opacity 1s ease-in;
}

.ReactModal__Overlay {
  opacity: 0;
  transition: opacity 500ms ease-in;
  outline: none;
}

.ReactModal__Overlay--after-open {
  opacity: 1;
  transition: opacity 1s ease-in;
  outline: none;
}

.ReactModal__Overlay--before-close {
  opacity: 0;
  transition: opacity 500ms ease-in;
  outline: none;
}

.closeModal {
  float: right;
  padding: none;
  padding-top: 2.2%;
  margin-right: 11px;
  margin-top: -5px;
  font-family: Helvetica, sans-serif;
  color: darkgray;
  cursor: pointer;
  outline: none;
  transition: opacity 500ms ease-in;
}

.closeModal:hover {
  float: right;
  padding: none;
  padding-top: 2.2%;
  margin-right: 11px;
  margin-top: -5px;
  font-family: Helvetica, sans-serif;
  color: black;
  cursor: pointer;
  outline: none;
  transition: opacity 500ms ease-in;
}
