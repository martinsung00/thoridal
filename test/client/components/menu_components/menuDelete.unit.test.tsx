import React from "react";
import renderer from "react-test-renderer";
import { Delete } from "../../../../src/client/components/menu_components/MenuDelete";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

describe("Menu Option Update Tests", function () {
  it("should render hello world", function () {
    const container = renderer.create(<Delete />).toJSON();
    expect(container).toMatchSnapshot();
  });

  it("should open the modal on click & close the modal on clicking the close button", function () {
    const wrapper = shallow(<Delete />);

    // Simulate opening the modal
    wrapper.find("#menu-delete").simulate("click");
    expect(wrapper).toMatchSnapshot();

    // Simulate closing the modal
    wrapper.find("#close-modal").simulate("click");
    expect(wrapper).toMatchSnapshot();

    wrapper.unmount();
  });
});
