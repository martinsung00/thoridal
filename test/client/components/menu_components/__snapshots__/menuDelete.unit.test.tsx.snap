// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Option Update Tests should open the modal on click & close the modal on clicking the close button 1`] = `ShallowWrapper {}`;

exports[`Menu Option Update Tests should open the modal on click & close the modal on clicking the close button 2`] = `ShallowWrapper {}`;

exports[`Menu Option Update Tests should render hello world 1`] = `
<div>
  <div
    id="menu-delete"
    onClick={[Function]}
  >
    <div>
      Delete a Trade
    </div>
    <div />
    <div>
      Ã—
    </div>
  </div>
</div>
`;
